File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
    int n=nums.size();
    int i=0;
    int j;
    for(j=0;j<n;j++){
        if(nums[j]==0) k--;
        if(k<0 and nums[i++]==0) k++;
    }
    return j-i;
    }
};

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var longestOnes = function(nums, k) {
 let i=0;
    let j;
    for(j=0;j<nums.length;j++){
        if(nums[j]==0) k--;
        if(k<0 && nums[i++]==0) k++;
    }
    return j-i;
};